app/routers.py

```diff
@@ -33,11 +33,94 @@ def http_exception_handler(request: Request, e: HTTPException) -> JSONResponse:
             e,
         )
 
-        # Return the JSON response for v2 compatibility
-        if isinstance(e.detail, dict) and e.detail.get('errors'):
-            return JSONResponse(status_code=e.status_code, content=e.detail)
-        else:
-            raise e
+        error_type = 'HTTPException'
+
+        if e.status_code in (401, 403):
+            if e.detail == 'Not authenticated':
+                # special case to override status code and message to match v2
+                e.status_code = 401
+                e.detail = 'Unauthorized, authentication token must be provided'
+            error_type = 'AuthError'
+
+        errors = {'errors': [{'error': error_type, 'message': e.detail}], 'status_code': e.status_code}
+        return JSONResponse(status_code=e.status_code, content=errors)
+
+    @staticmethod
+    def request_validation_error_handler(request: Request, e: RequestValidationError) -> JSONResponse:
+        """Log and handle RequestValidationError errors.
+
+        Convert Pydantic/FastAPI errors structure to a v2 json response.
+
+        Args:
+            request (Request): The original request data.
+            e (RequestValidationError): The exception data.
+
+        Returns:
+            JSONResponse: The JSON response when validation errors are present.
+        """
+        status_code = status.HTTP_400_BAD_REQUEST
+
+        logger.warning(
```

Comment:
We don't log any validation errors as warnings in NAPI. These should be INFO logs since the service is operating as expected by rejecting bad requests.